app/main.py

```diff
@@ -1,3 +1,62 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: any) -> any:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: any) -> any:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        return self
+
+    def __mul__(self, other: any) -> any:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: any) -> any:
+        if isinstance(other, (int, float)) and other != 0:
+            return Distance(round(self.km / other, 2))
+        elif other == 0:
+            raise ZeroDivisionError("Division by zero is not allowed")
+        return NotImplemented
+
+    def __lt__(self, other: any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km < other
+        elif isinstance(other, Distance):
+            return self.km < other.km
+
+    def __gt__(self, other: any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km > other
+        elif isinstance(other, Distance):
+            return self.km > other.km
+
+    def __eq__(self, other: any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km == other
+        elif isinstance(other, Distance):
+            return self.km == other.km
+
+    def __le__(self, other: any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km <= other
+        elif isinstance(other, Distance):
+            return self.km <= other.km
+
+    def __ge__(self, other: any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km >= other
+        elif isinstance(other, Distance):
+            return self.km >= other.km
```

Comment:
Ensure that the `__ge__` method returns `NotImplemented` if `other` is not a `Distance`, `int`, or `float`, to maintain consistency with the other comparison methods.